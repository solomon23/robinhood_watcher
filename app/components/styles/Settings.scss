@import 'Variables';
$white: #aaa;

:global {
  @import "~muicss/lib/sass/mui.scss";
}

:global {
  .mui-select > select {
    color: $white
  }

  .mui-textfield > input {
    color: $white;

    &::placeholder {
      color: $white;
    }
  }

  .mui-textfield > label {
    color: $white
  }

  .mui-select__menu {
    color: #000;
  }

  .mui-select:focus > select {
    border-color: $green;
  }

  .mui-select > select:focus {
    border-color: $green;
  }

  .mui-select:focus > label, .mui-select > select:focus ~ label {
    color: $green;
  }

  .mui-textfield > input:focus {
    border-color: $green;
  }

  .mui-textfield > input:focus ~ label {
    color: $green;
  }
}

.dropdown {
  label {
    color: $white
  }
}

.inputs {
  margin: 60px 20px 20px 20px;

}

.buttons {
  text-align: center;
  margin: 16px 28px 0 28px;

  button {
    width: 100%;
    display: inline-block;
    background-color: $red;
    color: #fff;

    &:focus, &:hover {
      background-color: darken($color: $red, $amount: 10);
      color: darken($color: #fff, $amount: 10);
    }
  }

  .submit button {
    background-color: $green;

    &:focus, &:hover {
      background-color: darken($color: $green, $amount: 10);
      color: darken($color: #fff, $amount: 10);
    }
  }
}
