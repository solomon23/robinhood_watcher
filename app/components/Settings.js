// @flow
import React, { Component } from 'react'
import Option from 'muicss/lib/react/option'
import Select from 'muicss/lib/react/select'
import Input from 'muicss/lib/react/input'
import Button from 'muicss/lib/react/button'
import styles from './styles/Settings.scss'

type Props = {
  saveSettings: (UserSettings) => void,
  quit: () => void,
  logout: () => void,
  settings: UserSettings
}

type State = {
  updated: boolean,
  settings: UserSettings
}

export default class Settings extends Component<Props, State> {
  props: Props

  constructor(props: Props) {
    super(props)

    this.onSaveClick = this.onSaveClick.bind(this)
    this.onQuitClick = this.onQuitClick.bind(this)
    this.onLogoutClick = this.onLogoutClick.bind(this)
    this.onRefreshChange = this.onRefreshChange.bind(this)

    this.state = {
      updated: false,
      settings: { ...props.settings },
    }
  }

  componentWillReceiveProps(next: Props) {
    const { updated } = this.state
    if (!updated) {
      this.setState({ settings: next.settings })
    }
  }

  /* :: onSaveClick: Function */
  onSaveClick() {
    const { saveSettings } = this.props
    const { settings } = this.state

    saveSettings({ ...settings })
  }

  /* :: onQuitClick: Function */
  onQuitClick() {
    const { quit } = this.props

    quit()
  }

  /* :: onLogoutClick: Function */
  onLogoutClick() {
    const { logout } = this.props

    logout()
  }

  /* :: onRefreshChange: Function */
  onRefreshChange(e: SyntheticInputEvent<HTMLInputElement>) {
    const { settings: currentSettings } = this.state
    const settings = { ...currentSettings, refreshInterval: parseInt(e.target.value) }
    this.setState({ settings })
  }

  render() {
    const { settings } = this.state

    return (
      <div>
        <div className={styles.inputs}>
          <Select label="Refresh every..." onChange={this.onRefreshChange} value={settings.refreshInterval} className={styles.dropdown} >
            <Option value={1} label="1 Minute" />
            <Option value={2} label="2 Minutes" />
            <Option value={5} label="5 Minutes" />
            <Option value={10} label="10 Minutes" />
            <Option value={15} label="15 Minutes" />
          </Select>
          <Select label="View Change By" value="percent" className={styles.dropdown}>
            <Option value="percent" label="Percent Change" />
            <Option value="gain/loss" label="Individual Gain/Loss" />
            <Option value="total-gain/loss" label="Total Gain/Loss" />
          </Select>
          <Input label="Notify me if a prices moves by..." placeholder="2%" pattern="[0-9]" />
        </div>
        <div className={styles.buttons}>
          <div className={styles.submit}>
            <Button onClick={this.onSaveClick} type="button">Submit</Button>
          </div>
          <div>
            <Button type="button" onClick={this.onLogoutClick}>Logout</Button>
          </div>
          <div>
            <Button type="button" onClick={this.onQuitClick}>Quit</Button>
          </div>
        </div>
    </div>
    )
  }
}
