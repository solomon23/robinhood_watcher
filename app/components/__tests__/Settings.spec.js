import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import Settings from '../Settings'

Enzyme.configure({ adapter: new Adapter() })

let actions = null

const defaultState = {
  settings: { refreshInterval: 1, viewChangeBy: 'TOTAL', notifyPercent: 10 },
}

describe('Settings component', () => {
  beforeEach(() => {
    actions = { saveSettings: jest.fn(), quit: jest.fn(), logout: jest.fn() }
  })

  it('should display', () => {
    const component = shallow(<Settings {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should change refresh', () => {
    const component = shallow(<Settings {...defaultState} {...actions} />)
    component.find('Select').at(0).simulate('change', { target: { value: '10' } })
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should view by', () => {
    const component = shallow(<Settings {...defaultState} {...actions} />)
    component.find('Select').at(1).simulate('change', { target: { value: 'PERCENT' } })
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should fire submit', () => {
    const component = shallow(<Settings {...defaultState} {...actions} />)
    component.find('Select').at(1).simulate('change', { target: { value: 'PERCENT' } })
    component.find('.submit').childAt(0).simulate('click')
    expect(actions.saveSettings).toHaveBeenCalledWith({ notifyPercent: 10, refreshInterval: 1, viewChangeBy: 'PERCENT' })
  })

  it('should fire quit', () => {
    const component = shallow(<Settings {...defaultState} {...actions} />)
    component.find('Button').at(2).simulate('click')
    expect(actions.quit).toHaveBeenCalled()
  })

  it('should fire logout', () => {
    const component = shallow(<Settings {...defaultState} {...actions} />)
    component.find('Button').at(1).simulate('click')
    expect(actions.logout).toHaveBeenCalled()
  })
})
