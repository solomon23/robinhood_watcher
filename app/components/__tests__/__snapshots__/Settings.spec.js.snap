// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component should change refresh 1`] = `
<div>
  <div
    className="inputs"
  >
    <Select
      className="dropdown"
      label="Refresh every..."
      name=""
      onChange={[Function]}
      onClick={null}
      onKeyDown={null}
      placeholder={null}
      readOnly={false}
      useDefault={false}
      value={10}
    >
      <Option
        className=""
        label="1 Minute"
        value={1}
      />
      <Option
        className=""
        label="2 Minutes"
        value={2}
      />
      <Option
        className=""
        label="5 Minutes"
        value={5}
      />
      <Option
        className=""
        label="10 Minutes"
        value={10}
      />
      <Option
        className=""
        label="15 Minutes"
        value={15}
      />
    </Select>
    <Select
      className="dropdown"
      label="View Change By"
      name=""
      onChange={[Function]}
      onClick={null}
      onKeyDown={null}
      placeholder={null}
      readOnly={false}
      useDefault={false}
      value="TOTAL"
    >
      <Option
        className=""
        label="Percent Change"
        value="PERCENT"
      />
      <Option
        className=""
        label="Individual Gain/Loss"
        value="INDIVIDUAL"
      />
      <Option
        className=""
        label="Total Gain/Loss"
        value="TOTAL"
      />
    </Select>
    <_class
      className=""
      floatingLabel={false}
      hint={null}
      invalid={false}
      label="Notify me if a prices moves by..."
      pattern="[0-9]"
      placeholder="2%"
    />
  </div>
  <div
    className="buttons"
  >
    <div
      className="submit"
    >
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Submit
      </Button>
    </div>
    <div>
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Logout
      </Button>
    </div>
    <div>
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Quit
      </Button>
    </div>
  </div>
</div>
`;

exports[`Settings component should display 1`] = `
<div>
  <div
    className="inputs"
  >
    <Select
      className="dropdown"
      label="Refresh every..."
      name=""
      onChange={[Function]}
      onClick={null}
      onKeyDown={null}
      placeholder={null}
      readOnly={false}
      useDefault={false}
      value={1}
    >
      <Option
        className=""
        label="1 Minute"
        value={1}
      />
      <Option
        className=""
        label="2 Minutes"
        value={2}
      />
      <Option
        className=""
        label="5 Minutes"
        value={5}
      />
      <Option
        className=""
        label="10 Minutes"
        value={10}
      />
      <Option
        className=""
        label="15 Minutes"
        value={15}
      />
    </Select>
    <Select
      className="dropdown"
      label="View Change By"
      name=""
      onChange={[Function]}
      onClick={null}
      onKeyDown={null}
      placeholder={null}
      readOnly={false}
      useDefault={false}
      value="TOTAL"
    >
      <Option
        className=""
        label="Percent Change"
        value="PERCENT"
      />
      <Option
        className=""
        label="Individual Gain/Loss"
        value="INDIVIDUAL"
      />
      <Option
        className=""
        label="Total Gain/Loss"
        value="TOTAL"
      />
    </Select>
    <_class
      className=""
      floatingLabel={false}
      hint={null}
      invalid={false}
      label="Notify me if a prices moves by..."
      pattern="[0-9]"
      placeholder="2%"
    />
  </div>
  <div
    className="buttons"
  >
    <div
      className="submit"
    >
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Submit
      </Button>
    </div>
    <div>
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Logout
      </Button>
    </div>
    <div>
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Quit
      </Button>
    </div>
  </div>
</div>
`;

exports[`Settings component should view by 1`] = `
<div>
  <div
    className="inputs"
  >
    <Select
      className="dropdown"
      label="Refresh every..."
      name=""
      onChange={[Function]}
      onClick={null}
      onKeyDown={null}
      placeholder={null}
      readOnly={false}
      useDefault={false}
      value={1}
    >
      <Option
        className=""
        label="1 Minute"
        value={1}
      />
      <Option
        className=""
        label="2 Minutes"
        value={2}
      />
      <Option
        className=""
        label="5 Minutes"
        value={5}
      />
      <Option
        className=""
        label="10 Minutes"
        value={10}
      />
      <Option
        className=""
        label="15 Minutes"
        value={15}
      />
    </Select>
    <Select
      className="dropdown"
      label="View Change By"
      name=""
      onChange={[Function]}
      onClick={null}
      onKeyDown={null}
      placeholder={null}
      readOnly={false}
      useDefault={false}
      value="PERCENT"
    >
      <Option
        className=""
        label="Percent Change"
        value="PERCENT"
      />
      <Option
        className=""
        label="Individual Gain/Loss"
        value="INDIVIDUAL"
      />
      <Option
        className=""
        label="Total Gain/Loss"
        value="TOTAL"
      />
    </Select>
    <_class
      className=""
      floatingLabel={false}
      hint={null}
      invalid={false}
      label="Notify me if a prices moves by..."
      pattern="[0-9]"
      placeholder="2%"
    />
  </div>
  <div
    className="buttons"
  >
    <div
      className="submit"
    >
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Submit
      </Button>
    </div>
    <div>
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Logout
      </Button>
    </div>
    <div>
      <Button
        className=""
        color="default"
        onClick={[Function]}
        size="default"
        type="button"
        variant="default"
      >
        Quit
      </Button>
    </div>
  </div>
</div>
`;
